### PR TITLE
Add custom validation message to international phone number form validation

### PR DESCRIPTION
**Pull request by kuzmany**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL |  https://github.com/mautic/developer-documentation/pull/109
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Based on quick fix https://github.com/mautic/mautic/pull/6645 and feature added to 2.14.1 https://github.com/mautic/mautic/pull/5400 we add option to change validation message for  nternational phone number form validation. 

This PR also add extend addValidator event with parameter formType, which allow extend validation tab by developers (see docs https://github.com/mautic/developer-documentation/pull/109)

This is example how developers can add options to validator tab and then validate based on setting: https://github.com/mautic/mautic/pull/6650/files#diff-e68b0c198b71013d6f317456841c2abd

image

#### Steps to test this PR:
1: Create form
2. Create Phone field. See properties tab and option: Validate as internationally-standardized format. Enable it and also setup validation message
3. Try send form with non validated form nubmer (xxx for example)
4. You should see your custom message
5. Then go to phone field. And remove custom validation text message
6. Try send form with non validated form nubmer (xxx for example)
7. You should see default validation message

### CODE REVIEW

**Comment by escopecz** [comment]
@kuzmany as Norman pointed me to this PR from #6645, could you move the validation to the Validation tab as I suggested in [this comment](https://github.com/mautic/mautic/pull/6645#issuecomment-426424960)

I still thin that the default validation message should be simplified.

**Comment by npracht** [comment]
I'd suggest:
>Please use the following international phone number format [+][country code][subscriber number] for this field (eg: ‪+14028650000).

**Comment by Woeler** [comment]
I'm not getting through the validator with this imaginary, but true to standards Dutch phone number: ‪+31612345678. I am also getting this instead of the error message `mautic.form.submission.phone.invalid`.

File: app/bundles/FormBundle/EventListener/FormValidationSubsriber.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+     * @param Events\ValidationEvent $event
+     */
+    public function onFormValidate(Events\ValidationEvent $event)
+    {
+        $field = $event->getField();
+        if ($field->getType() === 'tel' && !empty($field->getValidation()['international'])) {
+            if (!empty($field->getValidation()['international_validationmsg'])) {
+                $event->failedValidation($field->getValidation()['international_validationmsg']);
+            } else {
+                $event->failedValidation('mautic.form.submission.phone.invalid');
```
Comment (Woeler) [inline app/bundles/FormBundle/EventListener/FormValidationSubsriber.php]: Translation service seems to be missing

File: app/bundles/FormBundle/Translations/en_US/messages.ini [hunk 2]
```diff
@@ -103,7 +103,7 @@ mautic.form.field.type.pagebreak="Page break"
 mautic.form.field.type.radiogrp="Radio group"
 mautic.form.field.type.textarea="Textarea"
 mautic.form.field.type.file="File"
-mautic.form.field.type.tel.international="Validate as internationally-standardized format"
+mautic.form.field.type.tel.international="Validate as internationally format"
```
Comment (Woeler) [inline app/bundles/FormBundle/Translations/en_US/messages.ini]: "Validate as international format"

Comment (npracht) [reply app/bundles/FormBundle/Translations/en_US/messages.ini]: Done

File: app/bundles/FormBundle/EventListener/FormValidationSubsriber.php (hunk 1 above)
Comment (kuzmany) [reply app/bundles/FormBundle/EventListener/FormValidationSubsriber.php]: Good catch. Fixed

File: app/bundles/FormBundle/EventListener/FormValidationSubsriber.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+namespace Mautic\FormBundle\EventListener;
+
+use Mautic\CoreBundle\EventListener\CommonSubscriber;
+use Mautic\FormBundle\Event as Events;
+use Mautic\FormBundle\FormEvents;
+
+/**
+ * Class FormValidationSubsriber.
+ */
+class FormValidationSubsriber extends CommonSubscriber
```
Comment (Woeler) [inline app/bundles/FormBundle/EventListener/FormValidationSubsriber.php]: Class name has a typo `FormValidationSubscriber`

**Comment by kuzmany** [comment]
@Woeler should fixed now. Tested on mautibox

**Comment by npracht** [comment]
@Woeler everything is fine for you now ?

**Comment by Woeler** [comment]
Mautibox is currently not working with this pr.

**Comment by kuzmany** [comment]
@Woeler @npracht Merge to actual staging, hope helps

File: app/bundles/FormBundle/EventListener/FormValidationSubsriber.php (hunk 1 above)
Comment (dsp76) [reply app/bundles/FormBundle/EventListener/FormValidationSubsriber.php]: @kuzmany it seems to ignore a custom message and always output "Please use the following international phone number format [+][country code][subscriber number] for this field (eg: ‪+14028650000)."

I just added my own validation error text and it was ignored in 2.15.3.

**Pull request merged**